addons/crm/models/res_users.py

```diff
@@ -0,0 +1,14 @@
+from odoo import _, api, models
+
+
+class ResUsers(models.Model):
+    _inherit = "res.users"
+
+    @api.depends_context('crm_formatted_display_name_team')
+    def _compute_display_name(self):
+        super()._compute_display_name()
+        team_id = int(self.env.context.get('crm_formatted_display_name_team', 0))
```

Comment:
Well no, you should not receive strings at this point ^^ context should be considered as containing valid values (aka integers here).